crates/agglayer-telemetry/src/traces.rs

```diff
@@ -0,0 +1,201 @@
+use std::time::Duration;
+
+use agglayer_config::tracing::{TracingFormat, TracingOutput};
+use anyhow::anyhow;
+use opentelemetry::{global, trace::TracerProvider, KeyValue};
+use opentelemetry_otlp::WithExportConfig;
+use opentelemetry_sdk::{
+    propagation::TraceContextPropagator,
+    trace::{BatchConfigBuilder, BatchSpanProcessor, Sampler, SpanLimits},
+    Resource,
+};
+use tracing_subscriber::{prelude::*, util::SubscriberInitExt, EnvFilter};
+
+pub fn setup_tracing(config: &agglayer_config::TracingConfig, version: &str) -> anyhow::Result<()> {
+    let mut layers = Vec::new();
+
+    for writer in &config.outputs {
+        // Setup instrumentation if both otlp agent url and
+        // otlp service name are provided as arguments
+        if writer == &TracingOutput::Otlp {
```

Comment:
Maybe we should move the `otlp` configuration out of this loop, or even have a dedicated `TracesConfig` struct that have a method that build and export the layer directly